sdk/js/src/EventEmitter.ts

```diff
@@ -93,8 +93,9 @@ export class EventEmitter {
                         variable.callback?.call(variable, variable.value)
                     })
                 }
```

Comment:
do you think there's a way to type the subscribe function better, so that when subscribing to the variableUpdated events it would tell you to expect either a `Variable` or `null`?

Reply:
i've been trying a bunch of things but there's only one subscribe method and the type of `eventHandler` will be different depending on the first argument to subscribe, so think it would have to be something like:

```
type HandlerTypes = {
    VARIABLE_UPDATED: (key: string, variable: DVCVariable<DVCVariableValue> | null) => void,
    ...
}
```
`    subscribe<T extends EventNames>(key: T, handler: Pick<HandlerTypes, T>): void {`
to try to match the string key to the expected type 

but the problem is `EventNames` would have to match the actual regex strings for this to work, ie:
```
type EventNames = 
| `variableUpdated:${string}` 
| `newVariables:${string}` 
| `featureUpdated:${string}` 
| 'initialized' 
| 'error'
```
otherwise it won't accept the regex matching for the key param

... but then i can't index the HandlerTypes type, because i only have these dynamic types like `variableUpdated:${string}`  and not something fixed like VARIABLE_UPDATED

Reply:
wait never mind i got it working with function overloading

Reply:
yup overloading is what you want